### ISSUE TITLE
Only imperial temp sensors are exposed

### ISSUE THREAD

**Issue by OnFreund** [home-assistant-libs/aioshelly#7]
Since the metric and imperial temperature sensors have the same description (`'D': 'extTemp'`), they overwrite when creating `sensor_ids`:
 
https://github.com/home-assistant-libs/aioshelly/blob/e60429b75b24486641b8a5616bab6e491e1ddd69/aioshelly/__init__.py#L216

**Comment by balloob** [comment]
Meh, that is annoying that it's duplicate description :(

**Comment by OnFreund** [comment]
Yeah, and they're exposed even when you don't have a temperature sensor connected 🤷‍♂️ 
Maybe the client can pass in the desired temperature unit?

**Comment by balloob** [comment]
No. This API wrapper needs to expose just the values from the API, no logic to choose or convert things. That's up to the application that uses this lib.

**Comment by OnFreund** [comment]
ok, then maybe:
```
self.sensor_ids = {(val["D"], val["U"]): val["I"] for val in sensors.values()} 
```
And the sensor code in HA will have to adapt.

**Comment by balloob** [comment]
Not really a big fan because it means that you now need to know the unit. 

Maybe we should get rid of the sensor IDs shortcut and instead have two new methods `get_sensor(description)` and `get_sensors(description)`. Then `get_sensor(description)` will raise if there are multiple matches.

We would then have to get rid of the `block.<attribute>` access, and instead return "value classes" which contain value, unit, description. They will all be linked to the device so are updated when the device updates.

**Comment by OnFreund** [comment]
But then you also need to know if you're expecting one or multiple (we know `extTemp` is multiple but there could be others, and maybe in some models `extTemp` is single).
Maybe only have `get_sensors(description)`, and return a list of length 1 in case it's single?

**Comment by balloob** [comment]
that's possible too.

**Comment by chemelli74** [comment]
> Yeah, and they're exposed even when you don't have a temperature sensor connected 🤷‍♂️

True in the "cit/d" call, but won't be there in the "cit/s".
And that's the same for many other values, I can create a list if needed.

I would say that we should do it the other way around: wait for a "cit/s" and from there create the real device sensors vs "cit/d".

Simone

**Comment by chemelli74** [comment]
Taking into account that all sensors should handle the channel ("L"), only the following 2 descriptions overlap:

- deviceTemp
- extTemp

Simone

**Comment by balloob** [comment]
Fix in #9

**Comment by OnFreund** [comment]
👍 
I also think we should consider @chemelli74's comment. We can avoid redundant entities (that don't even report a value) in HA.
